src/d3d12/d3d12_model_pool.hpp

```diff
@@ -78,11 +86,26 @@ namespace wr
 		MemoryBlock* AllocateMemory(MemoryBlock* start_block, std::size_t size, std::size_t alignment);
 		void FreeMemory(MemoryBlock* heap_start_block,  MemoryBlock* block);
 		
-		std::queue<internal::MeshInternal*> m_mesh_stage_queue;
+		std::vector<std::queue<internal::MeshInternal*>> m_mesh_stage_queues;
+
+		std::uint32_t m_current_queue;
 
 		std::uint64_t m_vertex_buffer_size;
 		std::uint64_t m_index_buffer_size;
 
+		util::ThreadPool* m_thread_pool;
+
+		std::vector<d3d12::CommandList*> m_command_lists;
+		std::vector<d3d12::CommandList*> m_transition_command_lists;
```

Comment:
Unused?

Reply:
I'm submitting the copy commands on the copy queue, but I can't do transitions there. I'm using two arrays of command lists for this reason.